controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -221,52 +224,74 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
 
 	// check request status already contains the name of the resource. if not, add it.
 	if r.StorageClassRequest.Spec.Type == "blockpool" {
+		// initialize in-memory structs
+		r.cephRadosNamespace = &rookCephv1.CephBlockPoolRadosNamespace{}
+		r.cephRadosNamespace.Namespace = r.OperatorNamespace
 		r.cephBlockPool = &rookCephv1.CephBlockPool{}
 		r.cephBlockPool.Namespace = r.OperatorNamespace
-		for _, res := range r.StorageClassRequest.Status.CephResources {
-			if res.Kind == "CephBlockPool" {
-				r.cephBlockPool.Name = res.Name
-				break
-			}
+
+		// check if a CephBlockPoolRadosNamespace resource exists for the desired storageconsumer and storageprofile.
+		cephRadosNamespaceList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+		err := r.list(
+			cephRadosNamespaceList,
+			client.InNamespace(r.OperatorNamespace),
+			client.MatchingFields{util.OwnerUIDIndexName: string(r.StorageClassRequest.UID)})
+		if err != nil {
+			return err
+		}
+
+		// if we found no CephBlockPoolRadosNamespaces, generate a new name
+		// if we found only one CephBlockPoolRadosNamespace with our query, we're good
+		// if we found more than one CephBlockPoolRadosNamespace, we can't determine which one to select, so error out
+		rnsItemsLen := len(cephRadosNamespaceList.Items)
+		if rnsItemsLen == 0 {
+			rnsNewName := fmt.Sprintf("cephradosnamespace-%s-%s", r.StorageClassRequest.Spec.StorageProfile, r.storageConsumer.Name)
```

Comment:
Using hashed names is annoying, since it just makes it harder for humans to actually do meaningful troubleshooting, though I understand the technical limitations that lead to us using them. I believe that I had done the character counting such that it would take an almost max-length profile name string to cause issues, but I'm not about to do that calculation again. I'll just hash the names. 

And the empty-name edge case is a holdover from before you told me to gut the CephBlockPool reconciliation, as there would always be a StorageProfile defined by this point. Fixed.